MetadataMigration/src/main/java/org/opensearch/migrations/clusters/RemoteCluster.java

```diff
@@ -0,0 +1,68 @@
+package org.opensearch.migrations.clusters;
+
+
+
+import org.opensearch.migrations.DataFiltersArgs;
+import org.opensearch.migrations.Version;
+import org.opensearch.migrations.VersionMatchers;
+import org.opensearch.migrations.metadata.GlobalMetadataCreator;
+import org.opensearch.migrations.metadata.IndexCreator;
+import org.opensearch.migrations.metadata.tracing.RootMetadataMigrationContext;
+
+import com.rfs.common.OpenSearchClient;
+import com.rfs.common.http.ConnectionContext;
+import com.rfs.models.GlobalMetadata;
+import com.rfs.models.IndexMetadata;
+import com.rfs.version_os_2_11.GlobalMetadataCreator_OS_2_11;
+import com.rfs.version_os_2_11.IndexCreator_OS_2_11;
+import com.rfs.version_remote.RemoteIndexMetadataFactory;
+import com.rfs.version_remote.RemoteMetadataFactory;
+import lombok.RequiredArgsConstructor;
+import lombok.ToString;
+
+@RequiredArgsConstructor
+@ToString(onlyExplicitlyIncluded = true)
+public class RemoteCluster implements TargetCluster, SourceCluster {
```

Comment:
There is a question of where is the registry of potential source/target providers can live, should it be inside this class controlled via a central registration progress see 005c2c9d60d6d4d85c504c28bd364d1967156e7f, or should there be more clustering around SourceResourceProvider implementations.

Reply:
I've replaced TargetResourceProvider with the ClusterProviderRegistry which now centrally handles the set of snapshot providers or remote providers.